merlin/dag/schema_mixin.py

```diff
@@ -0,0 +1,192 @@
+#
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import annotations
+
+from enum import Flag, auto
+
+from merlin.dag.selector import ColumnSelector
+from merlin.schema import ColumnSchema, Schema
+
+
+class Supports(Flag):
+    """Indicates what type of data representation this operator supports for transformations"""
+
+    # cudf dataframe
+    CPU_DATAFRAME = auto()
+    # pandas dataframe
+    GPU_DATAFRAME = auto()
+    # dict of column name to numpy array
+    CPU_DICT_ARRAY = auto()
+    # dict of column name to cupy array
+    GPU_DICT_ARRAY = auto()
+
+
+class ComputeSchemaMixin:
```

Comment:
This mixin allows non-operator classes to use the same schema computation machinery as operator without having to become operators and implement the full operator interface.